src/hotspot/share/gc/g1/g1CollectionSet.hpp

```diff
@@ -150,11 +146,13 @@ class G1CollectionSet {
   volatile uint _collection_set_cur_length;
   uint _collection_set_max_length;
 
+  uint _eden_region_length;
+  uint _survivor_region_length;
+
+  G1CollectionSetRegionList _initial_old_regions;
```

Comment:
Why is the whole list saved in the field? I'd expect initial-old-regions is a transient list used to move regions from candidate list to cset (live only inside `G1CollectionSet::finalize_old_part`).

`_initial_old_regions` and `_optional_old_regions` share some similarity on  the name, but semantically, it's closer to eden/survior regions, so sth like `uint _initial_old_region_length;`.

Reply:
I do not have a too strong opinion either way, so I'll change it (back).